relay-server/src/actors/global_config.rs

```diff
@@ -253,25 +251,55 @@ impl Service for GlobalConfigService {
         tokio::spawn(async move {
             let mut shutdown_handle = Controller::shutdown_handle();
 
+            let config = &self.config;
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match (
+                config.relay_mode(),
+                config.has_credentials(),
+                config.global_config(),
+            ) {
+                (RelayMode::Managed, true, global_config) => {
+                    if global_config.is_some() {
+                        relay_log::info!("ignoring static global config in managed mode");
+                    }
+                    relay_log::info!("serving global configs fetched from upstream");
+                    // This request will trigger the request intervals when internal_rx receives the
+                    // result from upstream.
+                    Self::request_global_config(self.upstream.clone(), self.internal_tx.clone());
+                }
+                (RelayMode::Managed, false, _) => {
+                    // NOTE(iker): not making a request results in the sleep handler
+                    // not being reset, so no new requests are made.
+                    relay_log::info!("serving default global configs due to lacking credentials");
+                }
+                (
+                    RelayMode::Proxy | RelayMode::Static | RelayMode::Capture,
+                    _,
+                    Some(global_config),
+                ) => {
+                    relay_log::info!("serving static global config loaded from file");
+                    self.global_config_watch.send(global_config.clone()).ok();
+                }
+                (RelayMode::Proxy | RelayMode::Static | RelayMode::Capture, _, None) => {
+                    relay_log::info!(
+                        "serving default global configs due to lacking static global config file"
```

Comment:
not a big fan of the implicit nature here, Id prefer we create the watch here, but then we'd have to pass it into `handle_message` and `handle_upstream_result`.